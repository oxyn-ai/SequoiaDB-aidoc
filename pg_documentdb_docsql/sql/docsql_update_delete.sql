CREATE EXTENSION IF NOT EXISTS pg_documentdb;
CREATE EXTENSION IF NOT EXISTS pg_documentdb_docsql;

SELECT documentdb_docsql.create_database('docdb2');
SELECT documentdb_docsql.create_collection('docdb2','orders');

INSERT INTO docsql_docdb2.orders(document) VALUES
('{ "_id": "o1", "user": "u1", "total": 10 }'),
('{ "_id": "o2", "user": "u2", "total": 20 }');

UPDATE docsql_docdb2.orders
SET document = '{ "_id": "o1", "user": "u1", "total": 15 }'
WHERE (document->>'_id') = 'o1';

DELETE FROM docsql_docdb2.orders
WHERE (document->>'_id') = 'o2';

SELECT (document->>'_id') id, (document->>'total') total
FROM docsql_docdb2.orders
ORDER BY id;
