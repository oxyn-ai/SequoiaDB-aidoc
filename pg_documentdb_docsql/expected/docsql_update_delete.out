CREATE EXTENSION IF NOT EXISTS pg_documentdb;
CREATE EXTENSION IF NOT EXISTS pg_documentdb_docsql;
SELECT documentdb_docsql.create_database('docdb2');
 create_database 
-----------------
 
(1 row)

SELECT documentdb_docsql.create_collection('docdb2','orders');
 create_collection 
-------------------
 
(1 row)

INSERT INTO docsql_docdb2.orders(document) VALUES
('{ "_id": "o1", "user": "u1", "total": 10 }'),
('{ "_id": "o2", "user": "u2", "total": 20 }');
INSERT 0 2
UPDATE docsql_docdb2.orders
SET document = '{ "_id": "o1", "user": "u1", "total": 15 }'
WHERE (document->>'_id') = 'o1';
UPDATE 1
DELETE FROM docsql_docdb2.orders
WHERE (document->>'_id') = 'o2';
DELETE 1
SELECT (document->>'_id') id, (document->>'total') total
FROM docsql_docdb2.orders
ORDER BY id;
 id | total 
----+-------
 o1 | 15
(1 row)
